### PR TITLE
Add `__repr__` for `class Job`

### PR DESCRIPTION
**Pull request by vtomole**
Fixes: https://github.com/SupertechLabs/cirq-superstaq/issues/97

### CODE REVIEW

File: cirq_superstaq/service.py [hunk 1]
```diff
@@ -77,7 +76,7 @@ def counts_to_results(
     return result
 
 
-class Service:
+class Service(finance.Finance, logistics.Logistics, user_config.UserConfig):
```
Comment (richrines1) [inline cirq_superstaq/service.py:79]: is this from #123?

Comment (vtomole) [reply cirq_superstaq/service.py]: Yes. I've merged `main` into this branch to get rid of that diff.

File: cirq_superstaq/job.py [hunk 2]
```diff
@@ -169,3 +169,11 @@ def counts(
...
     def __str__(self) -> str:
         return f"Job with job_id={self.job_id()}"
+
+    def __repr__(self) -> str:
+        return f"cirq_superstaq.Job(client={self._client!r}, job_dict={self._job!r})"
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(self, type(other)):
+            return False
+        return self._job == other._job
```
Comment (richrines1) [inline cirq_superstaq/job.py:179]: is there any reason we'd want to check `self._client == other._client` as well?

Comment (vtomole) [reply cirq_superstaq/job.py:179]: I think this is good enough of a comparison because the backend should never duplicate `job_ids`

**Review by richrines1 — APPROVED** [review]
lgtm

**Pull request merged**